src/App.tsx

```diff
@@ -15,48 +17,108 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+const SORT_BY = {
+  ALPHA: 'alpha',
+  LENGTH: 'length',
+};
+
+type SortingParams = {
+  sortBy: string,
+  reverse: boolean
+};
+
+function getVisibleGoods(
+  goods:string[],
+  {
+    sortBy,
+    reverse,
+  }: SortingParams,
+): string[] {
+  const result = [...goods];
+
+  if (sortBy) {
+    result.sort((a, b) => {
+      switch (sortBy) {
+        case SORT_BY.ALPHA:
+          return a.localeCompare(b);
+        case SORT_BY.LENGTH:
+          return a.length - b.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverse) {
+    result.reverse();
+  }
+
+  return result;
+}
+
 export const App: React.FC = () => {
+  const [sortBy, setSortBy] = useState('');
+  const [reverse, setReverse] = useState(false);
+
+  const visibleGoods = getVisibleGoods(goodsFromServer,
+    { sortBy, reverse });
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={cn('button is-info', {
+            'is-light': sortBy !== SORT_BY.ALPHA,
+          })}
+          onClick={() => setSortBy(SORT_BY.ALPHA)}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={cn('button is-success', {
+            'is-light': sortBy !== SORT_BY.LENGTH,
+          })}
+          onClick={() => setSortBy(SORT_BY.LENGTH)}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={cn('button is-warning ', {
+            'is-light': !reverse,
+          })}
+          onClick={() => setReverse(!reverse)}
         >
           Reverse
         </button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {(sortBy || reverse) && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => {
+              setSortBy('');
+              setReverse(false);
+            }}
```

Comment:
Move to separate function  (transfer the function to a variable, and here only write the name of the function)